kingpin/actors/rightscale/server_array.py

```diff
@@ -996,6 +1002,19 @@ def _wait_until_healthy(self, array, sleep=60):
             if min_count <= count:
                 raise gen.Return()
 
+            all_instances = yield self._client.get_server_array_current_instances(array)
+            stranded_instances = yield self._client.get_server_array_current_instances(
```

Comment:
These two queries along with the "operational" request above constitute for 3 queries for the same info. While it might be somewhat simpler to do it this way it seems wrong to query RightScale with 3 API calls for something that can be done with just 1.